opencart 1.5.6.4/upload/admin/controller/module/smaily_for_opencart.php

```diff
@@ -122,7 +201,47 @@ public function index() {
         );
         $this->response->setOutput($this->render());
     }
-
+    protected function handleCustomerSync() {
+        $isRequestMethodPost = $this->request->server['REQUEST_METHOD'] === 'POST';
+        $hasPermissionsToModify = $this->user->hasPermission('modify', 'module/smaily_for_opencart');
+        if (!$isRequestMethodPost || !$hasPermissionsToModify) {
+            return;
+        }
+        $response = [];
+        // Load Smaily admin model for saving settings.
+        $this->load->model('smailyforopencart/admin');
+        // Declare customer sync objects.
+        $subscribe_status = $this->request->post['smaily_for_opencart_enable_subscribe'];
+        // For if no customer sync fields are selected.
+        if (!empty($this->request->post['smaily_for_opencart_syncronize_additional'])) {
+            $subscribe_additional = $this->request->post['smaily_for_opencart_syncronize_additional'];
+        } else {
+            $subscribe_additional = array();
+        }
+        $subscribe_sync_token = $this->request->post['smaily_for_opencart_sync_token'];
+        
+        // Data validation.
+        $subscribe_status = ($subscribe_status == 0 || $subscribe_status == 1) ? $subscribe_status : 0;
+        // Flipping array allows to use isset for validation.
+        $subscribe_additional = (
+            isset(array_flip($subscribe_additional)['firstname']) || 
+            isset(array_flip($subscribe_additional)['lastname']) ||
+            isset(array_flip($subscribe_additional)['telephone']) ||
+            isset(array_flip($subscribe_additional)['date_added'])) ? $subscribe_additional : array();
+        // Validate and sanitize cron token field.
+        $subscribe_sync_token = !empty($subscribe_sync_token) ? $this->db->escape($subscribe_sync_token) : uniqid();
+        // Add declared objects to array.
+        $settings = [
+          'subscribe_status' => ($subscribe_status),
+          'subscribe_additional' => ($subscribe_additional),
+          'subscribe_sync_token' => ($subscribe_sync_token),
+        ];
+        // Save customer sync settings to db.
+        $this->model_smailyforopencart_admin->editSettingValue('smaily', 'smaily_customer_sync', $settings);
+        $this->session->data['success'] = $this->language->get('text_success');
+        $this->redirect($this->url->link('extension/module', 'token=' . $this->session->data['token'], 'SSL'));
+        return;
```

Comment:
Returning here is not really necessary, because there is not code after `$this->redirect`.